crates/matrix-sdk-base/src/latest_event.rs

```diff
@@ -50,26 +51,33 @@ pub enum PossibleLatestEvent<'a> {
 /// Decide whether an event could be stored as the latest event in a room.
 /// Returns a LatestEvent representing our decision.
 #[cfg(feature = "e2e-encryption")]
-pub fn is_suitable_for_latest_event(event: &AnySyncTimelineEvent) -> PossibleLatestEvent<'_> {
+pub fn is_suitable_for_latest_event<'a>(
+    event: &'a AnySyncTimelineEvent,
+    prev_latest: Option<&EventId>,
+) -> PossibleLatestEvent<'a> {
+    use ruma::events::room::message::Relation;
+
     match event {
         // Suitable - we have an m.room.message that was not redacted or edited
         AnySyncTimelineEvent::MessageLike(AnySyncMessageLikeEvent::RoomMessage(message)) => {
             // Check if this is a replacement for another message. If it is, ignore it
             if let Some(original_message) = message.as_original() {
-                let is_replacement =
-                    original_message.content.relates_to.as_ref().map_or(false, |relates_to| {
-                        if let Some(relation_type) = relates_to.rel_type() {
-                            relation_type == RelationType::Replacement
+                // We include messages that are not replacements; but we allow a replacement if
+                // that's the replacement to the previous known latest event.
+                let is_suitable =
+                    original_message.content.relates_to.as_ref().map_or(true, |relates_to| {
```

Comment:
If I read this right, it is equivalent to:

```
let relates_to = original_message.content.relates_to().as_ref();
let is_suitable = if let Some(relates_to) = relates_to {
    // This is a relation: is it replacement of the previous latest event?
    if let Relation::Replacement(c) = &relates_to {
        // It is a replacement: suitable if it replaces the latest event
        prev_latest == Some(&c.event_id)
    else {
        // Some other type of relation: not suitable
        false
    }
} else {
    // Not a relation: definitely suitable
    true
};
```

I find the original version really dense - what do you think about using a more expanded version like the above? I might even make it a separate function...